src/generic/Matrix.jl

```diff
@@ -58,6 +58,11 @@ Base.isassigned(a::Union{Mat,MatRingElem}, i, j) = isassigned(a.entries, i, j)
 #
 ################################################################################
 
+function similar(x::MatSpaceElem{T}, r::Int, c::Int) where T <: NCRingElement
+   M = Matrix{T}(undef, (r, c))
+   return MatSpaceElem{T}(base_ring(x), M)
+end
```

Comment:
This method would go if we agree on the proposal to forbid `MatSpaceElem{T}` instances if `dense_matrix_type(T) != MatSpaceElem{T}`.

But in the meantime it is necessary to preserve Nemo tests.